provisioner/provisioner.go

```diff
@@ -289,12 +298,20 @@ func sendEventOrIgnore(pvcName, pvName, capacity, stgType, method string) {
 		stgType = "nfs-" + stgType
 	}
 
-	analytics.New().Build().ApplicationBuilder().
-		SetVolumeType(stgType, method).
-		SetDocumentTitle(pvName).
-		SetCampaignName(pvcName).
+	analytics.New().CommonBuild(GetEngineName(stgType, method)).ApplicationBuilder().
+		SetVolumeName(pvName).
+		SetVolumeClaimName(pvcName).
 		SetLabel(analytics.EventLabelCapacity).
-		SetReplicaCount("", method).
+		SetAction(DefaultUnknownReplicaCount).
 		SetCategory(method).
 		SetVolumeCapacity(capacity).Send()
 }
+
+// Wrapper for setting the default storage-engine for volume-provision event
+func GetEngineName(volType, method string) string {
+	if method == analytics.VolumeProvision && volType == "" {
+		return DefaultCASType
+	} else {
+		return volType
+	}
+}
```

Comment:
The function GetEngineName isn't required IMO. Instead of using `GetEngineName(stgType, method)`, you can directly use stgType, it'd be the same I think